oak/server/oak_runtime.cc

```diff
@@ -46,37 +46,40 @@ asylo::Status OakRuntime::Initialize(const ApplicationConfiguration& config) {
 
   // TODO: Support creating multiple Nodes and Channels connecting them.
   const WebAssemblyNode& web_assembly_node = node_configuration.web_assembly_node();
-  node_ = OakNode::Create(web_assembly_node.module_bytes());
-  if (node_ == nullptr) {
+  auto node = OakNode::Create(web_assembly_node.module_bytes());
+  if (node == nullptr) {
     return asylo::Status(asylo::error::GoogleError::INVALID_ARGUMENT, "Failed to create Oak Node");
   }
 
-  SetUpChannels();
+  // Setup the default channels for the node.
+  SetUpChannels(*node.get());
+
+  // TODO: This needs to be configurable
+  // Create the pseudo node required.
+  grpc_node_ = OakGrpcNode::Create();
+  SetUpGrpcChannels(*node.get());
+
+  // Move ownership to vector.
+  wasm_nodes_.push_back(std::move(node));
 
   return asylo::Status::OkStatus();
 }
-grpc::Service* OakRuntime::GetGrpcService() { return node_.get(); }
-
-asylo::Status OakRuntime::StartCompletionQueue(std::unique_ptr<grpc::AsyncGenericService> service,
-                                               std::unique_ptr<grpc::ServerCompletionQueue> queue) {
-  // Use the serivce and queue provided.
-  // TODO: check to see if we already started this.
-  module_service_ = std::move(service);
-  completion_queue_ = std::move(queue);
 
-  // Start a new thread to process the gRPC completion queue.
-  std::thread thread(&OakRuntime::CompletionQueueLoop, this);
-  thread.detach();
+asylo::Status OakRuntime::Start() {
+  LOG(INFO) << "Starting runtime";
+  for (auto& node : wasm_nodes_) {
```

Comment:
Is the order of starting the nodes important? Even if not, it would be useful to leave a comment here.

Reply:
It's not right now. I think we should aim for not having any dependencies on the ordering of nodes